examples/seq2seq/create_student.py

```diff
@@ -0,0 +1,124 @@
+    12: {1: [11], 2: [5, 11], 3: [3, 7, 11], 6: [1, 3, 5, 8, 10, 11]},
+    16: {1: [15], 4: [4, 9, 12, 15], 8: [1, 3, 5, 7, 9, 11, 13, 15]},
+    6: {1: [5], 2: [3, 5], 3: [1, 4, 5], 4: [1, 2, 4, 5]},
+    2: {1: [1], 2: [0, 1]},
+}
+
+
+def get_layers_to_copy(n_student, n_teacher):
+    try:
+        val = LAYERS_TO_COPY[n_teacher][n_student]
+        # assert len(LAYERS_TO_SUPERVISE[n_teacher][n_student]) == len(val) == n_student
+        return val
+    except KeyError:
+        if n_student != n_teacher:
+            warnings.warn(
+                f"no hardcoded layers to copy for teacher {n_teacher} -> student {n_student}, defaulting to first {n_student}"
+            )
+        return list(range(n_student))
+
+
+def copy_to_student(
+    d_layers_to_copy, e_layers_to_copy, student_encoder_layers, student_decoder_layers, student, teacher
+):
+    different_encoder: bool = student_encoder_layers != teacher.config.encoder_layers
+    different_decoder = student_decoder_layers != teacher.config.decoder_layers
+    if different_decoder:
+        copy_layers(teacher.model.decoder.layers, student.model.decoder.layers, d_layers_to_copy)
+    if different_encoder:
+        copy_layers(teacher.model.encoder.layers, student.model.encoder.layers, e_layers_to_copy)
+
+
+def init_student(student, teacher):
+    teacher_state_dict = teacher.state_dict()
+    info = student.load_state_dict(teacher_state_dict, strict=False)
+    assert info.missing_keys == [], info.missing_keys
+    return student, info
+
+
+def copy_decoder_layers(teacher, student, l2copy=[0, 2, 4, 7, 9, 11]):
+    copy_layers(teacher.model.decoder.layers, student.model.decoder.layers, l2copy)
+
+
+def copy_layers(teacher_layers: nn.ModuleList, student_layers: nn.ModuleList, layers_to_copy: List) -> None:
+    layers_to_copy = nn.ModuleList([l for i, l in enumerate(teacher_layers) if i in layers_to_copy])
+    assert len(student_layers) == len(layers_to_copy), f"{len(student_layers)} != {len(layers_to_copy)}"
+    student_layers.load_state_dict(layers_to_copy.state_dict())
+
+
+def create_student(teacher, student_encoder_layers, student_decoder_layers):
+    teacher = AutoModelForSeq2SeqLM.from_pretrained(teacher).eval()
+    student_updates = {
+        "decoder_layers": student_decoder_layers,
+        "encoder_layers": student_encoder_layers,
+    }
+    e_layers_to_copy: List = get_layers_to_copy(student_updates["encoder_layers"], teacher.config.encoder_layers)
+    d_layers_to_copy: List = get_layers_to_copy(student_updates["decoder_layers"], teacher.config.decoder_layers)
+
+    kw = teacher.config.to_diff_dict()
+    kw.update(student_updates)
+    # Copy weights
+    student_cfg = teacher.config_class(**kw)
+    student = type(teacher)(student_cfg)
+    student, _ = init_student(student, teacher)
+    copy_to_student(
+        d_layers_to_copy, e_layers_to_copy, student_encoder_layers, student_decoder_layers, student, teacher
+    )
+    return student
+
+
+def main():
```

Comment:
You just need this and an import, right?

Reply:
One should be able to import and call `create_student` function which will return the student.